src/System.Text.Json/tests/Utf8JsonReaderTests.TryGet.cs

```diff
@@ -1202,6 +1203,26 @@ public static void TestingStringsConversionToGuid(string testString, string expe
             Assert.Equal(json.BytesConsumed, json.CurrentState.BytesConsumed);
         }
 
+        [Theory]
+        [MemberData(nameof(JsonGuidTestData.ValidGuidTests), MemberType = typeof(JsonGuidTestData))]
+        public static void TryGetGuid_HasValueSequence_RetrievesGuid(string testString, string expectedStr)
+        {
+            byte[] dataUtf8 = Encoding.UTF8.GetBytes($"\"{testString}\"");
+            ReadOnlySequence<byte> sequence = JsonTestHelper.GetSequence(dataUtf8, 1);
+            var json = new Utf8JsonReader(sequence, isFinalBlock: true, state: default);
+
+            Guid expected = new Guid(expectedStr);
+
+            Assert.True(json.Read(), "Read string value");
```

Comment:
The message if this assert fails isn't that helpful. Remove?

Reply:
I'll give opposing feedback: Add messages to the other Assert.True or Assert.False calls.  The distinct messages say which thing failed.

```C#
Assert.True(json.Read(), "json.Read()");
...
Assert.True(json.HasValueSequence, "json.HasValueSequence");
Assert.False(json.ValueSequence.IsEmpty, "json.ValueSequence.IsEmpty");
Assert.True(json.TryGetGuid(out Guid actual), "TryGetGuid");
...
```

So the message becomes something like

```
TryGetGuid
  Expected: True
    Actual: False

    at [yada yada]
```

With no message it just says something like

```
Assert.True
  Expected: True
    Actual: False

  at [yada yada]
```

And since there are three different Assert.True calls currently, it's hard to see what failed (particularly important it it becomes intermittent).

Reply:
Agreed - each assert ideally is 100% unique to the method so it's that much easier to find the LOC when investigating a failure.

Reply:
> so it's that much easier to find the LOC when investigating a failure.

Easier than the actual LOC shown in the call stack?

> Assert.True(json.HasValueSequence, "json.HasValueSequence");

So, the purpose of the message here is just to identify what line failed? I thought the reason you would provide a string is to give additional context/info that might help. Just duplicating the assert condition itself seemed unnecessary to me.

I get the value in more context specific info, for example:
https://github.com/dotnet/corefx/blob/30c7f263fc60a558cbabf388555ebea211bce642/src/System.Text.Json/tests/Utf8JsonReaderTests.cs#L605

> So the message becomes something like

Oh... I see now. Yes, that would be useful. Thanks!